### DISCUSSION TITLE
Difference in the route of google maps and organic maps from destination A to destination B

### DISCUSSION THREAD

**Discussion by sharmaachintya08** [organicmaps/organicmaps#5036] [Ideas]
I searched for the same destination from my home town on organic maps and google maps and this is what I got: 

**country** -- **India**
**Dest. A** -- **Dest. B**
**Delhi** -- **Leh, Kashmir**

organic maps :
 
![omaps](https://user-images.githubusercontent.com/80171589/234223473-3459446c-28e3-4524-a6e4-172ac0684c77.png)

Google maps : 

![car_rouote_to_le](https://user-images.githubusercontent.com/80171589/234223441-b8a65537-b6de-4a32-9260-77492ce2c164.png)

As you can see there are almost 20 hours more in the route given by organic maps.

Why is that ? 
do we need to improve our routing algorithms for better routes or is it because of some of the missing places or destinations in the open maps?

**Comment by Jean-BaptisteC** [comment]
Why did you have select car on Organic Maps and motorcycle on Google Maps ?

Check this routing https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=28.65172%2C77.22194%3B34.16343%2C77.58434#map=6/31.447/77.234

**Reply by sharmaachintya08** [reply]
Since there is no motorbike option on organic maps, I had to go with the car one. But in this case, the route for car and bike is the same in google maps. But i should update it to car anyway.

**Reply by biodranik** [reply]
Did you check the maps data quality in your area in OpenStreetMap.org ? If roads are wrongly mapped there, then our routing won't work properly. Good news is that you can always easily fix roads in OpenStreetMap.org yourself. For example, you can specify the road type, surface, speed limit, etc. Or even draw a missing road.

**Reply by sharmaachintya08** [reply]
Wow, that is cool. It is possible that the roads may be wrongly mapped or maybe some roads may be missing on OpenStreetMap because it's a very remote area, So maybe that is the reason for the longer route.

**Comment by leyan** [comment]
The destination shown is clearly different. In Google Maps, the destination is south of Amritsar shown on the left, while on OrganicMaps, Leh is much farther north, and behind a 3500m peak. I cannot reproduce your Google Maps screenshot, searching for Leh gives a result placed like OrganicMaps. 

And trying to reproduce your result in OrganicMaps gives me a 20h route, did you add some info on OpenStreetMap in the meantime?

**Comment by muralito** [comment]
Hi. Also there seems to be a data problem in the area that results in what seems not the best route.
This way is tertiary and has no maxspeed tag https://www.openstreetmap.org/way/293750680
(verified just before this posting)
The parallel trunk NH3 has maxspeed=50 so the algorithm prefers the tertiary road, because it has a faster speed in their default settings.

This is a problem that is seen when you set the maxspeed of a road to a speed slower than the default, and there are near roads with faster speeds (the default chosen by the algorithm).

Neither of the roads have errors in their tagging if looking at them one by one, but when you look the dataset like the router does, the problem appears. IMHO in this case it should be fixed by tagging the appropriate maxspeed in the tertiary road.

**Reply by dev24126** [reply]
Just a hint: Be sure that you have downloaded all the maps in between. If a map is missing this area will be circumnavigated.

**Reply by muralito** [reply]
> Just a hint: Be sure that you have downloaded all the maps in between. If a map is missing this area will be circumnavigated.

Sure, but how is that the missing area map related with the problem I described in OSM data?
